https://issues.redhat.com/browse/ACM-10095 Operator policy controller docs (#6173)
open-cluster-management/rhacm-docs: governance/policy_operator.adoc

[#policy-operator]
= Operator policy controller

> The operator policy controller allows you to monitor or install Operator Lifecycle Manager (OLM) operators across your clusters. You can monitor the health of various pieces of the operator and you can specify how to automatically handle updates to the operator. You can also distribute an operator policy to managed clusters by using the governance framework and adding the policy to the `policy-templates` field of a policy on the hub cluster.

Reviewer: Is it "or" or "and"? That is, do users need to choose which action or can they use both? If the latter, I would change to "and" : 

```suggestion
The operator policy controller allows you to monitor and install Operator Lifecycle Manager (OLM) operators across your clusters. You can monitor the health of various pieces of the operator and you can specify how to automatically handle updates to the operator. You can also distribute an operator policy to managed clusters by using the governance framework and adding the policy to the `policy-templates` field of a policy on the hub cluster.
```

> The operator policy controller allows you to monitor or install Operator Lifecycle Manager (OLM) operators across your clusters. You can monitor the health of various pieces of the operator and you can specify how to automatically handle updates to the operator. You can also distribute an operator policy to managed clusters by using the governance framework and adding the policy to the `policy-templates` field of a policy on the hub cluster.

Reviewer: "You can" can be redundant. What do you think of this rewrite that's a little more user action focused too : 

```suggestion
The operator policy controller allows you to monitor or install Operator Lifecycle Manager (OLM) operators across your clusters. Use the operator policy controller to monitor the health of various pieces of the operator and to specify how you want to automatically handle updates to the operator. You can also distribute an operator policy to managed clusters by using the governance framework and adding the policy to the `policy-templates` field of a policy on the hub cluster.
```

If the `remediationAction` for the operator policy is set to `enforce`, the controller creates resources on the target managed cluster to install the operator and approves updates based on the versions specified in the policy.

If the `remediationAction` set to `inform`, the controller only reports the status of the operator, including if any upgrades are available.

The primary field in an operator policy is `spec.subscription`. You must add information in the following fields to create an operator subscription:

- `channel`
- `name`
- `namespace`
- `source`
- `sourceNamespace` 

For more information, see the link:https://docs.openshift.com/container-platform/4.12/operators/understanding/olm/olm-understanding-olm.html#olm-subscription_olm-understanding-olm[Subscription] topic in the {ocp} documentation.

> You can use the `spec.versions` field in an operator policy to declare which versions of the operator are compliant. If the field is empty, any version running on the cluster is considered compliant. If you specify any of the fields, the version on the managed cluster must match one of the versions in the list for the policy to be compliant. If the policy is in `enforce` mode, it only approves updates to versions that are compliant.

Reviewer 2: ```suggestion
You can use the `spec.versions` field in an operator policy to declare which versions of the operator are compliant. If the field is empty, any version running on the cluster is considered compliant. If the field is not empty, the version on the managed cluster must match one of the versions in the list for the policy to be compliant. If the policy is in `enforce` mode, it only approves updates to versions that are compliant.
```

Saying "any of the fields" feels too vague, but I'm open to other options.

Author: That works too, thanks!

See the following operator policy sample:

[source,yaml]
----
apiVersion: policy.open-cluster-management.io/v1beta1
kind: OperatorPolicy
metadata:
  name: quay-policy
spec:
  remediationAction: inform
  severity: medium
  complianceType: musthave
  operatorGroup: [optional]
    name: scoped-operator-group
    namespace: quay-opns
    targetNamespaces:
      - quay-opns
  subscription:
    channel: stable-3.10
    name: quay-operator
    namespace: quay-opns
    installPlanApproval: Automatic
    source: redhat-operators
    sourceNamespace: openshift-marketplace
  versions:
    - quay-operator.v3.10.3

> ----

Reviewer 3: I recommend considering the additional resources section, if it is needed

Author: Added!

Reviewer 3: Thanks Oliver! i would also take the user back to the Policy controller intro since they were directed from there.

Author: I believe we agreed to not add the _return to_ links in additional resources. But we can bring this up to the team again

Reviewer 3: My understanding that we agreed to not return to the beginning of the same file. It's fine tho, not a big deal at all. If someone shouts about it we can easily add the link whenever. Thanks! I'm good with your current changes !

Author: OK, let's clarify in our channel though, maybe I misunderstood. I'll merge this for now, thanks!
